elastic/elasticsearch
# [buildkite] Add most of the remaining periodic pipelines

Brings over most of the rest of the periodic pipelines from Jenkins. The scripts, e.g. `eql-correctness.sh`, are straight from the Jenkins jobs.

- Migrate the following pipelines/jobs from Jenkins:
  - packaging-tests (unix, upgrade) (windows already migrated)
  - platform-support (unix, windows, arm)
  - bwc tests
  - concurrent-search-tests
  - encryption-at-rest (ear)
  - eql-correctness
  - example-plugins
  - release-tests
  - single-processor-node-tests
- Create a separate pipeline for packaging-tests, because of the large number of steps
- Create a separate pipeline for platform-support, because of the large number of steps
- Combine everything else into the generic "periodic" pipeline
- Use the new `branches.json` to trigger all of the periodic pipelines on the currently active branches
- Update Buildkite pipelines when running `updateCIBwcVersions`

The packaging-tests are green, but the rest of the pipelines fail... a lot. I believe, however, that the steps that fail and the frequencies line up with what we see in Jenkins at this point. For example, `release-tests` never passes.
